src/main/java/com/fasterxml/jackson/databind/cfg/JsonNodeFeature.java

```diff
@@ -79,7 +79,30 @@ public enum JsonNodeFeature implements DatatypeFeature
      *
      * @since 2.16
      */
-    FAIL_ON_NAN_TO_BIG_DECIMAL_COERCION(false)
+    FAIL_ON_NAN_TO_BIG_DECIMAL_COERCION(false),
+
+
+    /**
+     * Determines whether floating-point numbers should be deserialized into
+     * {@link java.math.BigDecimal} when reading {@link com.fasterxml.jackson.databind.JsonNode}s.
+     * This feature provides more precise control over number deserialization for {@code JsonNode}
+     * and takes precedence over {@link com.fasterxml.jackson.databind.DeserializationFeature#USE_BIG_DECIMAL_FOR_FLOATS}
+     * if explicitly set.
+     *
+     * <p>
+     * Behavior follows these rules:
+     * <ul>
+     *   <li>If explicitly enabled, floating-point numbers will be read as {@link java.math.BigDecimal}.</li>
+     *   <li>If explicitly disabled, floating-point numbers will be read as {@link java.lang.Double}.</li>
+     *   <li>If left undefined (default), the behavior follows {@code DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS}.</li>
+     * </ul>
+     *
+     * <p>
+     * Default value: undefined (relies on {@code DeserializationFeature.USE_BIG_DECIMAL_FOR_FLOATS}).
+     *
+     * @since 2.19
```

Comment:
I thought that was meant to default to undefined or null.

Reply:
I thought also we had something like `OptBoolean.DEFAULT` from JsonTypeInfo, but guess not 🤔